scipy/special/_basic.py

```diff
@@ -2036,13 +2036,18 @@ def factorial(n, exact=False):
     """
     if exact:
         if np.ndim(n) == 0:
-            return 0 if n < 0 else math.factorial(n)
+            if not np.isnan(n):
+                return 0 if n < 0 else math.factorial(n)
+            else:
+                return np.nan
```

Comment:
In this case return `n`. Reason: there are actually many NaNs and you want to propagate them if possible.

Reply:
ok-good point-I hadn't considered multiple NaNs scenarios.

Reply:
The `else:` is unneeded as both branches immediately return.  E.g. (also incorporating Josh's comment)
```
if np.ndim(n) == 0:
  if np.isnan(n):
    return n
 return 0 if n < 0 else math.factorial(n)
```

Reply:
@pvanmulbregt thanks for the feedback. I think with the `_factorial(n)` ufunc we no longer need the block. In other words: 
```
if np.ndim(n) == 0:
  if np.isnan(n):
    return n
 return 0 if n < 0 else math.factorial(n)
```
becomes:
```
if np.ndim(n) == 0:
    return _factorial(n)
```
Further simplifications in the other parts of the code were possible. 
I removed quite a bit of the top level `else:` block `nan` handling.